datasets/ett/ett.py

```diff
@@ -0,0 +1,242 @@
+            features=features,  # Here we define them above because they are different between the two configurations
+            # If there's a common (input, target) tuple from the features, uncomment supervised_keys line below and
+            # specify them. They'll be used if as_supervised=True in builder.as_dataset.
+            # supervised_keys=("sentence", "label"),
+            # Homepage of the dataset for documentation
+            homepage=_HOMEPAGE,
+            # License for the dataset if available
+            license=_LICENSE,
+            # Citation for the dataset
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        urls = _URLS[self.config.name]
+        filepath = dl_manager.download_and_extract(urls)
+
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                # These kwargs will be passed to _generate_examples
+                gen_kwargs={
+                    "filepath": filepath,
+                    "split": "train",
+                },
+            ),
+            datasets.SplitGenerator(
+                name=datasets.Split.TEST,
+                # These kwargs will be passed to _generate_examples
+                gen_kwargs={
+                    "filepath": filepath,
+                    "split": "test",
+                },
+            ),
+            datasets.SplitGenerator(
+                name=datasets.Split.VALIDATION,
+                # These kwargs will be passed to _generate_examples
+                gen_kwargs={
+                    "filepath": filepath,
+                    "split": "dev",
+                },
+            ),
+        ]
+
+    # method parameters are unpacked from `gen_kwargs` as given in `_split_generators`
+    def _generate_examples(self, filepath, split):
+        data = pd.read_csv(filepath, parse_dates=True, index_col=0)
+        start_date = data.index.min()
+
+        if self.config.name in ["m1", "m2"]:
+            factor = 4  # 15-min frequency
+        else:
+            factor = 1  # hourly frequency
+        train_end_date_index = 12 * 30 * 24 * factor  # 1 year
+
+        if split == "dev":
+            end_date_index = 12 * 30 * 24 + 4 * 30 * 24 * factor  # 1 year + 4 months
+        else:
+            end_date_index = 12 * 30 * 24 + 8 * 30 * 24 * factor  # 1 year + 8 months
+
+        if self.config.multivariate:
+            if split in ["test", "dev"]:
+                # rolling windows of prediction_length for dev and test
+                for i, index in enumerate(
+                    range(
+                        train_end_date_index,
+                        end_date_index,
+                        self.config.prediction_length,
+                    )
+                ):
+                    yield i, {
+                        "start": start_date,
+                        "target": data[: index + self.config.prediction_length].values.astype("float32").T,
```

Comment:
so yes it's confusing but it's right since this is for validation and testing where we just have a bigger end-index... (since the time series all start at the same time)

Reply:
makes sense, I should have noticed that from the previous datasets x)